File: Proyecto/02. Software/02. Back-End Java/workspace/sell.it.easy/src/main/java/com/sell/it/easy/entity/PedidoProductos.java
package com.sell.it.easy.entity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;

@Entity
public class PedidoProductos {

	@Id
	@GeneratedValue(strategy=GenerationType.AUTO)
	private Integer id;
	@ManyToOne(optional=false)
	private Producto producto;
	private Float precioCoste;
	private Float precioVentaPublico;
	@Column(nullable=false)
	private Float cantidad;
	
	@ManyToOne(optional=true)
	private Inventario inventario;
	
	@ManyToOne(optional=true)
	private Pedido pedido;
	
	
	public PedidoProductos(){
		
	}


	public Integer getId() {
		return id;
	}


	public void setId(Integer id) {
		this.id = id;
	}


	public Producto getProducto() {
		return producto;
	}


	public void setProducto(Producto producto) {
		this.producto = producto;
	}


	public Float getPrecioCoste() {
		return precioCoste;
	}


	public void setPrecioCoste(Float precioCoste) {
		this.precioCoste = precioCoste;
	}


	public Float getPrecioVentaPublico() {
		return precioVentaPublico;
	}


	public void setPrecioVentaPublico(Float precioVentaPublico) {
		this.precioVentaPublico = precioVentaPublico;
	}


	public Float getCantidad() {
		return cantidad;
	}


	public void setCantidad(Float cantidad) {
		this.cantidad = cantidad;
	}


	public Inventario getInventario() {
		return inventario;
	}


	public void setInventario(Inventario inventario) {
		this.inventario = inventario;
	}


	public Pedido getPedido() {
		return pedido;
	}


	public void setPedido(Pedido pedido) {
		this.pedido = pedido;
	}
	
	
}

File: Proyecto/02. Software/01. Front-End iOS/sell.it.easy_0.0.2/sell.it.easy/sell.it.easy/PedidoProductos.swift
//
//  PedidoProductos.swift
//  sell.it.easy
//
//  Created by David Márquez Delgado on 17/8/17.
//  Copyright © 2017 David Márquez Delgado. All rights reserved.
//

import Foundation
import ObjectMapper

class PedidoProductos: Mappable{
    /*
     
     @Id
     @GeneratedValue(strategy=GenerationType.AUTO)
     private Integer id;
     @ManyToOne(optional=false)
     private Producto producto;
     private Float precioCoste;
     private Float precioVentaPublico;
     @Column(nullable=false)
     private Float cantidad;
     
     @ManyToOne(optional=true)
     private Inventario inventario;
     
     @ManyToOne(optional=true)
     private Pedido pedido;
     */
    var id : Int?
    var producto : Producto!
    var precioCoste : Float?
    var precioVentaPublico : Float?
    var cantidad : Float!
    var inventario : Inventario?
    var pedido : Pedido?
    
    init(id: Int?, producto: Producto!, precioCoste: Float?, precioVentaPublico: Float?, cantidad: Float!, inventario: Inventario?, pedido: Pedido?) {
        self.id = id
        self.producto = producto
        self.precioCoste = precioCoste
        self.precioVentaPublico = precioVentaPublico
        self.cantidad = cantidad
        self.inventario = inventario
        self.pedido = pedido
    }
    
   
    required init?(map: Map) {}
    
    func mapping(map: Map) {
        id <- map["id"]
        producto <- map["producto"]
        precioCoste <- map["precioCoste"]
        precioVentaPublico <- map["precioVentaPublico"]
        cantidad <- map["cantidad"]
        inventario <- map["inventario"]
        pedido <- map["pedido"]
    }
}
